spec/unit/target/pod_target_spec.rb

```diff
@@ -182,6 +182,54 @@ module Pod
         @pod_target.build_product_path('$BUILT_PRODUCTS_DIR').should == '$BUILT_PRODUCTS_DIR/BananaLib/libBananaLib.a'
         @pod_target.scoped.first.build_product_path('$BUILT_PRODUCTS_DIR').should == '$BUILT_PRODUCTS_DIR/BananaLib-Pods/libBananaLib-Pods.a'
       end
+
+      it 'returns the correct header search paths' do
+        @pod_target.build_headers.add_search_path('BananaLib', Platform.ios)
+        @pod_target.sandbox.public_headers.add_search_path('BananaLib', Platform.ios)
+        header_search_paths = @pod_target.recursive_target_header_search_paths
+        header_search_paths.sort.should == [
+          '${PODS_ROOT}/Headers/Private',
+          '${PODS_ROOT}/Headers/Private/BananaLib',
+          '${PODS_ROOT}/Headers/Public',
+          '${PODS_ROOT}/Headers/Public/BananaLib',
+        ]
+      end
+
+      it 'returns the correct header search paths recursively for dependent targets' do
+        @pod_target.build_headers.add_search_path('BananaLib', Platform.ios)
+        @pod_target.sandbox.public_headers.add_search_path('BananaLib', Platform.ios)
+        @pod_target.sandbox.public_headers.add_search_path('monkey', Platform.ios)
+        monkey_spec = fixture_spec('monkey/monkey.podspec')
+        monkey_pod_target = PodTarget.new([monkey_spec], [@target_definition], config.sandbox)
+        monkey_pod_target.stubs(:platform).returns(Platform.ios)
+        @pod_target.stubs(:dependent_targets).returns([monkey_pod_target])
+        header_search_paths = @pod_target.recursive_target_header_search_paths
+        header_search_paths.sort.should == [
+          '${PODS_ROOT}/Headers/Private',
```

Comment:
I am still not sure if those should be included for any pod headers...I actually think they shouldn't but I am afraid it might cause a bit breakage. Perhaps I should TODO this?